model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/diffusion.py

```diff
@@ -542,44 +540,31 @@ def _do_diffusion_step(
 
         """
         klevels = self.grid.num_levels
-        cell_start_interior = self.grid.get_start_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.interior(dims.CellDim)
-        )
-        cell_start_nudging = self.grid.get_start_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.nudging(dims.CellDim)
-        )
-        cell_end_local = self.grid.get_end_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.local(dims.CellDim)
-        )
-        cell_end_halo = self.grid.get_end_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.halo(dims.CellDim)
-        )
+        cell_domain = h_grid.domain(dims.CellDim)
+        cell_start_interior = self.grid.start_index(cell_domain(h_grid.Zone.INTERIOR))
 
-        edge_start_nudging_plus_one = self.grid.get_start_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.nudging(dims.EdgeDim) + 1
-        )
-        edge_start_nudging = self.grid.get_start_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.nudging(dims.EdgeDim)
-        )
-        edge_start_lb_plus4 = self.grid.get_start_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.EdgeDim) + 4
-        )
-        edge_end_local = self.grid.get_end_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.local(dims.EdgeDim)
-        )
-        edge_end_local_minus2 = self.grid.get_end_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.local(dims.EdgeDim) - 2
-        )
-        edge_end_halo = self.grid.get_end_index(
-            dims.EdgeDim, h_grid.HorizontalMarkerIndex.halo(dims.EdgeDim)
+        cell_start_nudging = self.grid.start_index(cell_domain(h_grid.Zone.NUDGING))
+        cell_end_local = self.grid.end_index(cell_domain(h_grid.Zone.LOCAL))
+        cell_end_halo = self.grid.end_index(cell_domain(h_grid.Zone.HALO))
+
+        edge_domain = h_grid.domain(dims.EdgeDim)
+        edge_start_second_nudging_level = self.grid.start_index(
```

Comment:
rename to `edge_start_nudging_level_2` for consistency with the `Zone`?
```suggestion
        edge_start_nudging_level_2 = self.grid.start_index(
```